chain/pool/txpool.go

```diff
@@ -301,6 +301,21 @@ func (tp *TxnPool) GetTxnByCount(num int) (map[common.Uint256]*Transaction, erro
 	return txmap, nil
 }
 
+func (tp *TxnPool) GetNonceByTxnPool(addr common.Uint160) (uint64, error) {
+	if _, ok := tp.TxLists[addr]; !ok {
+		return 0, errors.New("no transactions in transaction pool")
```

Comment:
What about change it to `return 0, nil` because it's not an error, and most of the times this condition will be true.
Also after calling GetNonceByTxnPool and get returns in getNonceByAddr, the condition needs to be changed accordingly.

Reply:
most times this condition will be false. because if there is no transaction in txpool, this return nonce from this function is invalid.

Reply:
Oh actually I also meant most times this condition will be false. This will cause an warning error for every GetNonceByTxnPool call, which I consider as a false alarm because there is nothing wrong with it. The reasons I suggest changing it to `return 0, nil` are:
1. Conceptually this is not an error, it's just the return should be "empty"
2. 0 is the empty value here and should never be returned if tp.TxLists[addr] is not empty.
3. Those warning logs will be gone

Reply:
The return error can help to locate where return 0, because there are 2 positions to return 0. I can remove the log.warning in getNonceByAddr.

Reply:
If we remove log.warning, then if there is really an error returned by list.GetLatestNonce(), it will not be logged and we'd never know. What I suggest is that, we keep the error log if err is not nil, and we ignore the return value if returned value is zero OR error is not nil, then we can distinguish these two cases. Something like
```suggestion
txPoolNonce, err := txpool.GetNonceByTxnPool(pg)
if txPoolNonce == 0 {
	txPoolNonce = persistNonce
}
if err != nil {
 	log.Errorf("GetNonceByTxnPool error: %v", err)
 }
 if 
```

Reply:
the other error is  no transaction error too.

Reply:
OK I see